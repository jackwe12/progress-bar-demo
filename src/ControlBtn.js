import React from 'react';
import Button from 'react-bootstrap/Button';



const Btn =({buttonValue, index, handleScore, current, key})=>{
    return(
      
        (buttonValue > 0)
        ?(
        <span >
          <Button variant="secondary" onClick={()=>handleScore(current, buttonValue)}>+{buttonValue}</Button>
        </span>
        )
        :(
        <span >
          <Button variant="secondary" onClick={()=>handleScore(current, buttonValue)}>{buttonValue}</Button>
        </span>
        )
      
    );
}
export default Btn;