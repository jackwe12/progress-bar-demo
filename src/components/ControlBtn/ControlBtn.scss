.controlBtn-section{
    display: inline-block;
}
.bar-select-button{
    display: inline-block;
}
.btn-secondary{
    margin-left:1rem;
}