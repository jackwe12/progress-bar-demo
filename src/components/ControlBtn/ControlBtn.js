import React from 'react';
import Button from 'react-bootstrap/Button';
import PropTypes from 'prop-types';
import './ControlBtn.scss';


const Btn =({buttonValue, handleScore, current})=>{
    return(
      
        (buttonValue > 0)
        ?(
        <span >
          <Button variant="secondary" onClick={()=>handleScore(current, buttonValue)}>+{buttonValue}</Button>
        </span>
        )
        :(
        <span >
          <Button variant="secondary" onClick={()=>handleScore(current, buttonValue)}>{buttonValue}</Button>
        </span>
        )
      
    );
}
export default Btn;



Btn.propTypes = {
    buttonValue: PropTypes.number,
    handleScore: PropTypes.func,
    current: PropTypes.number,

}
