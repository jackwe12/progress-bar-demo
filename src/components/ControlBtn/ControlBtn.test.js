import ControlBtn from './ControlBtn';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

configure({ adapter: new Adapter() });

const mockHandler = jest.fn();
const wrapper = shallow(
    <ControlBtn
        buttonValue = {50}
        index = {0}
        handleScore = {mockHandler}
        current = {0}
        key={0}
    />);
const wrapper2 = shallow(
    <ControlBtn
        buttonValue = {-50}
        index = {0}
        handleScore = {mockHandler}
        current = {0}
        key={0}
    />);
const wrapper3 = mount(
    <ControlBtn
        buttonValue = {-50}
        index = {0}
        handleScore = {mockHandler}
        current = {0}
        key={0}
    />);

describe('Testing <ControlBtn/>', ()=>{
    test('snapshot testing',()=>{
        const tree =renderer
            .create(     <ControlBtn />)
            .toJSON();
            expect(tree).toMatchSnapshot();
    });
    test('buttonValue is positive',()=>{
        expect(wrapper.find('span').text()).toEqual('+50')
    });
    test('buttonValue is negative',()=>{
        expect(wrapper2.find('span').text()).toEqual('-50')
    });
    test('should call mock click Handler when btn clicked',()=>{
        const btn = wrapper3.find('button');
        btn.simulate('click');
        expect(mockHandler.mock.calls.length).toBe(1);
    });

});
