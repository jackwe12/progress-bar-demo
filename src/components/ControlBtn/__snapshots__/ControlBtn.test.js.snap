// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ControlBtn/> snapshot testing 1`] = `
<span>
  <button
    className="btn btn-secondary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
</span>
`;
